codalab/worker/local_run/docker_image_manager.py

```diff
@@ -102,13 +104,20 @@ def _cleanup(self):
                     entry_to_remove.digest,
                 )
                 try:
-                    image_to_delete = self._docker.images.get(entry_to_remove.id)
-                    tags_to_delete = image_to_delete.tags
-                    for tag in tags_to_delete:
-                        self._docker.images.remove(tag)
-                    # if we successfully removed the image also remove its cache entry
+                    # We try to delete images from image cache in LRU order
+                    self._docker.images.remove(entry_to_remove.id)
```

Comment:
From the docker rest api documentation, it seems that we can directly remove the docker image given image id provided.

Reply:
Neat!